src/shared/guards.ts

```diff
@@ -13,3 +22,23 @@ export const metadataMemberGuard = (
   input: MetadataMember | undefined | Partial<MetadataMember>
 ): input is MetadataMember =>
   input !== undefined && typeof input.fullName === 'string' && typeof input.type === 'string';
+
+export const isSdrFailure = (fileResponse: FileResponse): fileResponse is FileResponseFailure =>
```

Comment:
a few of these already exist in PDR and probably could ship in SDR if they'd be broadly useful.